### PR TITLE
Add `.. seealso::` in `Study.get_trials` and `Study.trials`

### PR DESCRIPTION
**Pull request by jmsykes83**
## Motivation

Add .. seealso:: in Study.get_trials and Study.trials to refer to each other https://github.com/optuna/optuna/issues/3578
Issue: https://github.com/optuna/optuna/issues/3578

## Description of the changes

Updated the dosctrings for Study.trials and Study.get_trials. To reference each other.

### CODE REVIEW

**Comment by nzw0301** [comment]
Could you apply my suggestions in https://github.com/optuna/optuna/pull/3860 except for replacing `--` with `::` ?

**Comment by jmsykes83** [comment]
> Could you apply my suggestions in #3860 except for replacing `--` with `::` ?
If you add it again.  It is no longer allowing me to commit suggested

File: optuna/study/study.py [hunk 1]
```diff
@@ -210,6 +210,9 @@ def trials(self) -> List[FrozenTrial]:
 
         Returns:
             A list of :class:`~optuna.trial.FrozenTrial` objects.
+            .. seealso::
+                See :func:`~optuna.study.Study.get_trails` for related method.
+
```
Comment (nzw0301) [inline optuna/study/study.py:215]:
```suggestion

            .. seealso::
                See :func:`~optuna.study.Study.get_trials` for related method.

```

File: optuna/study/study.py [hunk 2]
```diff
@@ -222,6 +225,8 @@ def get_trials(
         """Return all trials in the study.
 
         The returned trials are ordered by trial number.
+        .. seealso::
+            See :func:`~optuna.study.Study.trails` for related method.
```
Comment (nzw0301) [inline optuna/study/study.py:229]:
```suggestion
            See :attr:`~optuna.study.Study.trials` for related method.
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]
Fair enough. Can you merge these suggestions?

**Comment by nzw0301** [comment]
Let me rename the title for the readability of the forthcoming release note.

File: optuna/study/study.py [hunk 3]
```diff
@@ -222,6 +226,8 @@ def get_trials(
         """Return all trials in the study.
 
         The returned trials are ordered by trial number.
+        .. seealso::
```
Comment (nzw0301) [inline optuna/study/study.py:229]: I suppose one empty line is necessary to fix the CI error.

```suggestion

        .. seealso::
```

**Review by nzw0301 — APPROVED** [review]
Thanks! I'll merge this PR with the squash option

File: optuna/study/study.py [hunk 4]
```diff
@@ -223,6 +227,9 @@ def get_trials(
 
         The returned trials are ordered by trial number.
 
+        .. seealso::
+            See :attr:`~optuna.study.Study.trials` for related method.
```
Comment (nzw0301) [inline optuna/study/study.py:231]: Sorry, one more thing. Technically, property sounds more precise than `method`.

```suggestion
            See :attr:`~optuna.study.Study.trials` for related property.
```

**Pull request merged**